### PR TITLE
Bucketlist API implementation

### PR DESCRIPTION
**Pull request by Tschonti**
I think this should work, but I couldn't test it properly because the API calls to /achievement respond with almost no data (as if there were no categories or achievements), even though I can see mock data in the admin.  Is that feature not ready on the backend or did I set up the backend incorrectly? There are also some styling changes to be made.

### CODE REVIEW

**Review by triszt4n — COMMENTED** [review]
I guess there are no mocked data in the database yet, if you are interested in the Kotlin language, you could take a look at the `TestConfig.kt` file, where the mocked data is created. Otherwise you might be missing the `local,test` Spring profiles from your run config. Spring is a "natural" part of the Ultimate version of IntelliJ IDEA, you should take installing it into consideration (use your @sch.bme.hu account for free Student licence)

**Comment by Tschonti** [comment]
ooh no I messed something up with git

File: src/main/client/src/components/@pages/AchievementPage.tsx [hunk 1]
```diff
@@ -116,53 +116,61 @@ export const AchievementPage: React.FC<AchievementPageProps> = (props) => {
...
-        <Heading marginBottom="0px">{data.achievement.title}</Heading>
-        <AchievementStatusBadge status={data.status} fontSize="lg" />
-        <Stack>
-          <Skeleton height="20px" />
-          <Skeleton height="20px" />
-          <Skeleton height="20px" />
-        </Stack>
-        {data.status === achievementStatus.NOT_SUBMITTED ? (
-          <form>
+        <Heading marginBottom="0px">{achDetails.achievement?.title}</Heading>
```
Comment (berenteb) [inline src/main/client/src/components/@pages/AchievementPage.tsx:132]: marginBottom={0}

**Review by berenteb — CHANGES_REQUESTED** [review]
One minor thing.
I can't get this to work, might be a backend related problem or I missed a setting. I get an empty category even with categories added on the admin panel. Also, you are showing a skeleton even if an empty array of achievements is returned. This is misleading, I suggest writiong out a message for that.

**Comment by Tschonti** [comment]
You need to be REALLY logged in to make it work 😃 What worked for me is this:
You launch the app, delete the cookies, log in to the admin, give yourself a group (tankör), delete the cookies again, log in again and it'll work

**Comment by berenteb** [comment]
Oh yeah right, the individual bucketlist was not available! Will look into it tomorrow.

**Comment by triszt4n** [comment]
> ooh no I messed something up with git

Let's look into this problem in the afternoon, contact me on Slack thx

File: src/main/client/src/components/@pages/AchievementPage.tsx (hunk 1 above)
Comment (triszt4n) [reply src/main/client/src/components/@pages/AchievementPage.tsx:132]: Shorthand: `mb={0}`

**Comment by berenteb** [comment]
> > ooh no I messed something up with git
> 
> Let's look into this problem in the afternoon, contact me on Slack thx
I need help to test out the feature, please contact me about this meeting!

**Comment by triszt4n** [comment]
Did the rebase, as the #97 test was successful (all files remained intact after the rebasing, functionality is same)

**Review by triszt4n — APPROVED** [review]

**Comment by triszt4n** [comment]
@Tschonti is still designing the UI, so don't merge yet

**Comment by Tschonti** [comment]
Rebased to the current master, so now it can be tested with authContext. File upload size limit is 1 MB in development.

**Review by berenteb — APPROVED** [review]

**Pull request merged**